package/src/constructs/api-check.ts

```diff
@@ -17,16 +17,30 @@ enum BodyType {
   NONE = 'NONE'
 }
 
+export enum HttpRequestMethod {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  PATCH = 'PATCH',
+  HEAD = 'HEAD',
+  DELETE = 'DELETE',
+  OPTIONS = 'OPTIONS',
+}
+
 interface BasicAuth {
   username: string
   password: string
 }
 
 export interface Request {
   url: string,
-  method: string,
+  method: HttpRequestMethod,
```

Comment:
I would use a union literal type here, is much more ergonomic from a user perspective. Even Axios does it that way: https://github.com/axios/axios/blob/18c1710f244f54d5891a017714bb58f21f5039bd/index.d.ts#L191